src/main/java/dook/tasks/TaskList.java

```diff
@@ -0,0 +1,49 @@
+package dook.tasks;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import dook.DookException;
+
+
+/**
+ * The TaskList class contains the list of all created tasks.
+ */
+public class TaskList {
+
+    private ArrayList<Task> tasks;
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public int numOfTasks() {
+        return this.tasks.size();
+    }
+
+    public boolean isEmpty() {
+        return this.tasks.isEmpty();
+    }
+
+    public Task delete(int taskNumber) throws DookException, IOException {
+        if (taskNumber < 0 || taskNumber > numOfTasks()) {
+            throw new DookException("You don't have that many tasks");
+        }
+
+        Task removed = this.tasks.remove(taskNumber);
+        return removed;
+    }
+
+    public Task add(Task task) {
```

Comment:
Could it be a good idea to add some JavaDocs here just to remind the user that this method also returns the task?